### PR TITLE
Improving scan and adding option to keep servos on 

### PR DESCRIPTION
**Pull request by lhuth**
This PR fixes three issues:
1 The scanning had a bug that can cause the scan to stop if the TLU takes to long to start. Now we wait until all components are running
2 It is now possible to also scan with decreasing values 
3 The stages servo is switched off per default and will slowly slide down. Now the servo can be set to not be switched off

### CODE REVIEW

**Comment by schuetzepaul** [comment]
Thanks for pushing this. I have a few more things for the PIStage, but I'll add that on top of your changes then.

**Comment by lhuth** [comment]
> Thanks for pushing this. I have a few more things for the PIStage, but I'll add that on top of your changes then.

I've added your change :) - thanks for spotting

**Comment by schuetzepaul** [comment]
I'd have more ;-)

**Comment by lhuth** [comment]
Guess this is ready for reveiw? @schuetzepaul

**Comment by schuetzepaul** [comment]
Yes, it definitely is.

**Comment by schuetzepaul** [comment]
BTW: I'm seeing more failures of eudaq2 when using the scan feature. I'm not sure though whether this is related to the bug fixed here. I've noticed that currently our log is not stored to disk, but I'll come back with the log messages as soon as it happens again.

File: user/piStage/hardware/src/RotaController.cc [hunk 1]
```diff
@@ -389,11 +389,14 @@ bool RotaController::move(const char *szAxis, double target)
...
     } while (bFlag);
     LogInfo(std::string("Axis ") + std::string(szAxis) + std::string(" finished moving to ") + std::to_string(position));
-    servoState = FALSE;                   //uppercase because it comes from hardware (gcs) library
-    if (!PI_SVO(ID, szAxis, &servoState)) //Set Servo state
-    {
-        HardError("SVO failed  : could not turn off servo (motor)");
-        return false;
+
+    if(!keepServoOn){
+        servoState = FALSE;                   //uppercase because it comes from hardware (gcs) library
```
[the remark this answers is not in the document]
Comment (lhuth) [reply user/piStage/hardware/src/RotaController.cc:394]: Not mine, part of the code copied from the example drivers :)

File: gui/src/scanHelper.cc [hunk 2]
```diff
@@ -40,7 +40,7 @@ bool Scan::setupScan(std::string globalConfFile, std::string scanConfFile) {
             if(!m_scan_is_time_based && Counter == "wrongPara")
                 return scanError("To run a scan based on a number of events, \"eventCounter\" needs to be specified in section"+std::to_string(i));
             if(name == "wrongPara" || param == "wrongPara"
-               || start == -123456789 || stop == -123456789 || step == -123456789)
+               || start == -123456789.0 || stop == -123456789.0 || step == -123456789.0)
```
[the remark this answers is not in the document]
Comment (lhuth) [reply gui/src/scanHelper.cc:43]: > BTW: I'm seeing more failures of eudaq2 when using the scan feature. I'm not sure though whether this is related to the bug fixed here. I've noticed that currently our log is not stored to disk, but I'll come back with the log messages as soon as it happens again.

Not sure if this is the case :) But let me know if you have a starting point for /testing debugging

**Comment by lhuth** [comment]
> BTW: I'm seeing more failures of eudaq2 when using the scan feature. I'm not sure though whether this is related to the bug fixed here. I've noticed that currently our log is not stored to disk, but I'll come back with the log messages as soon as it happens again.

Not sure if this is the case :) But let me know if you have a starting point for /testing debugging

File: gui/src/scanHelper.cc [hunk 3]
```diff
@@ -29,9 +29,9 @@ bool Scan::setupScan(std::string globalConfFile, std::string scanConfFile) {
         if(scanConf->HasSection(std::to_string(i))) {
             scanConf->SetSection(std::to_string(i));
             bool nested     = scanConf->Get("nested",false) && m_allow_nested_scan;
-            double start    = scanConf->Get("start",-123456789);
-            double step     = scanConf->Get("step",-123456789);
-            double stop     = scanConf->Get("stop",-123456789);
+            double start    = scanConf->Get("start",-123456789.0);
```
[the remark this answers is not in the document]
Comment (lhuth) [reply gui/src/scanHelper.cc:32]: Yes - changed

**Comment by simonspa** [comment]
Looks good to me, @eyiliu let's merge this?

**Comment by eyiliu** [comment]
@lhuth , CI reports build error related to ambiguous function `abs`,  please fix it.

**Comment by lhuth** [comment]
fixed since 4 days, anything else or good to go?

**Pull request merged**